modules/l4/l4_input_local.c

```diff
@@ -27,10 +27,9 @@ void l4_input_register_port(uint8_t proto, rte_be16_t port, const char *next_nod
 	LOG(DEBUG, "l4_input_register_port: proto=%hhu port=%hu -> %s", proto, p, next_node);
 	switch (proto) {
 	case IPPROTO_UDP:
-		if (udp_edges[port] != MANAGEMENT)
+		if (udp_edges[p] != MANAGEMENT)
 			ABORT("next node already registered for udp port=%hu", p);
-		udp_edges[proto] = gr_node_attach_parent("ip_input_local", next_node);
-		gr_node_attach_parent("ip6_input_local", next_node);
+		udp_edges[p] = gr_node_attach_parent("l4_input_local", next_node);
```

Comment:
Could you preserve the port in big endian? Keep `p` just for logging the port number, but use `port` as offset in the udp_edges array. This allows lookup in l4_input_local_process without any rte_be_to_cpu_16() call.

```c
		if (udp_edges[port] != MANAGEMENT)
			ABORT("next node already registered for udp port=%hu", p);
		udp_edges[port] = gr_node_attach_parent("l4_input_local", next_node);
```